packages/ethereum/contracts/proxy/HoprStakingProxyForNetworkRegistry.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '../IHoprNetworkRegistryRequirement.sol';
+
+/**
+ * @dev Interface for staking contract
+ * source code at https://github.com/hoprnet/hopr-stake/tree/main/contracts
+ * staking v2 is deployed at https://blockscout.com/xdai/mainnet/address/0x2cDD13ddB0346E0F620C8E5826Da5d7230341c6E
+ */
+contract IStake {
+  function stakedHoprTokens(address _account) public view returns (uint256) {}
+
+  function isNftTypeAndRankRedeemed3(
+    uint256 nftTypeIndex,
+    uint256 boostNumerator,
+    address hodler
+  ) external view returns (bool) {}
+}
+
+/**
+ * @dev Proxy for staking (v2) contract, which an "HoprNetworkRegistry requirement" is implemented
+ */
+contract HoprStakingProxyForNetworkRegistry is IHoprNetworkRegistryRequirement, Ownable {
+  struct NftTypeAndRank {
+    uint256 nftType;
+    uint256 nftRank;
+  }
+
+  IStake public immutable STAKE_CONTRACT;
```

Comment:
Please document all attributes and events.

Reply:
Done